OpenRA.Game/Graphics/Util.cs

```diff
@@ -73,18 +73,39 @@ public static void FastCopyIntoSprite(Sprite dest, Bitmap src)
 
 			try
 			{
-				var data = dest.Sheet.GetData();
-				var dataStride = dest.Sheet.Size.Width * 4;
-				var x = dest.Bounds.Left * 4;
-				var width = dest.Bounds.Width * 4;
-				var y = dest.Bounds.Top;
+				var destData = dest.Sheet.GetData();
+				var destStride = dest.Sheet.Size.Width;
+				var width = dest.Bounds.Width;
 				var height = dest.Bounds.Height;
 
-				var bd = src.LockBits(src.Bounds(),
-					ImageLockMode.ReadWrite, src.PixelFormat);
-				for (var row = 0; row < height; row++)
-					Marshal.Copy(IntPtr.Add(bd.Scan0, row * bd.Stride), data, (y + row) * dataStride + x, width);
-				src.UnlockBits(bd);
+				var srcData = src.LockBits(src.Bounds(),
+					ImageLockMode.ReadWrite, PixelFormat.Format32bppArgb);
+
+				unsafe
```

Comment:
Note: this reverts one of the changes made by one of @RoosterDragon's perf prs, but I don't see any cross-platform way to do this otherwise (`PixelFormat.Format32bppPArgb` doesn't appear to work under mono).